pandas/core/groupby/generic.py

```diff
@@ -1863,15 +1864,144 @@ def _transform_general(self, func, engine, engine_kwargs, *args, **kwargs):
     """
     )
 
     @Substitution(klass="DataFrame", example=__examples_dataframe_doc)
     @Appender(_transform_template)
-    def transform(self, func, *args, engine=None, engine_kwargs=None, **kwargs):
-        return self._transform(
-            func, *args, engine=engine, engine_kwargs=engine_kwargs, **kwargs
+    def transform(
+        self,
+        func: None
+        | (Callable | str | list[Callable | str] | dict[str, NamedAgg]) = None,
+        *args,
+        engine: str | None = None,
+        engine_kwargs: dict | None = None,
+        **kwargs,
+    ) -> DataFrame:
+        if func is None:
+            # Convert named aggregations to dictionary format
+            transformed_func = self._named_agg_to_dict(*kwargs.items())
+            kwargs = {}
+            if isinstance(transformed_func, dict):
+                return self._transform_multiple_funcs(
+                    transformed_func,
+                    *args,
+                    engine=engine,
+                    engine_kwargs=engine_kwargs,
+                    **kwargs,
+                )
+        else:
+            if isinstance(func, list):
+                func = maybe_mangle_lambdas(func)
+                return self._transform_multiple_funcs(
+                    func, *args, engine=engine, engine_kwargs=engine_kwargs, **kwargs
+                )
+            else:
+                return self._transform(
+                    func, *args, engine=engine, engine_kwargs=engine_kwargs, **kwargs
+                )
+
+    def _transform_multiple_funcs(
+        self,
+        func: Any,
+        *args,
+        engine: str | None = None,
+        engine_kwargs: dict | None = None,
+        **kwargs,
+    ) -> DataFrame:
+        results = []
+
+        if isinstance(func, dict):
+            for name, named_agg in func.items():
+                column_name = named_agg.column
+                agg_func = named_agg.aggfunc
+                result = self._transform_single_column(
+                    column_name,
+                    agg_func,
+                    *args,
+                    engine=engine,
+                    engine_kwargs=engine_kwargs,
+                    **kwargs,
+                )
+                result.name = name
+                results.append(result)
+            output = concat(results, axis=1)
+        elif isinstance(func, list):
+            col_names = []
+            columns = (com.get_callable_name(f) or f for f in func)
+            func_pairs = zip(columns, func)
+            for idx, (name, func_item) in enumerate(func_pairs):
+                result = self._transform(
+                    func_item,
+                    *args,
+                    engine=engine,
+                    engine_kwargs=engine_kwargs,
+                    **kwargs,
+                )
+                results.append(result)
+                col_names.extend([(col, name) for col in result.columns])
+            output = concat(results, axis=1)
```

Comment:
```suggestion
            output = concat(results, ignore_index=True, axis=1)
```

Reply:
Doesn't the concatenation of results using `concat` with `axis=1` align the columns correctly without needing to reset the index. The original index of the DataFrame will be preserved right?

Reply:
The resulting `.index` should be preserved, but the since you're overwriting the `.columns` 2 lines down, we don't need `concat` to come up with the merged `.column` result and do less work by just returning a `RangeIndex`

Reply:
Yeah, makes sense, understood.